### ISSUE TITLE
How to quantize MobileNetV2 for deeplabV3+ ?

### ISSUE THREAD

**Issue by kanul** [tensorflow/tensorflow#19735]
**System information**
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:Source
- **TensorFlow version (use command below)**:1.8.0rc0
- **Python version**:2.7.12
- **Bazel version (if compiling from source)**:0.12.0
- **GCC/Compiler version (if compiling from source)**:5.4.0
- **CUDA/cuDNN version**:cuda-9.0/7.0
- **GPU model and memory**:GeForce GTX 1080/8105MiB
- **Phone**:xiaomi5 (Snapdragon 820)
- **Exact command to reproduce**:
bazel run --config=opt //tensorflow/contrib/lite/toco:toco --
--input_file=/external_home/data/model/deeplabv3_mnv2_pascal_train_aug/frozen_inference_graph.pb
--output_file=/external_home/data/model/deeplabv3_mnv2_pascal_train_aug/kanul.tflite
--inference_type=QUANTIZED_UINT8
--input_shape=1,513,513,3
--input_array=sub_7
--output_array=logits/semantic/BiasAdd

**Describe the problem**
I have tried to quantize MobileNetV2 for deeplabV3+ with TFlite. But I fail to convert the model.
From the following issue, I saw that the operations were not supported for the option of quantization.

https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/model_zoo.md
Checkpoint name: mobilenetv2_coco_voc_trainaug

Who can explain and support to resolve the issue?

**Source code / logs**
bazel run --config=opt //tensorflow/contrib/lite/toco:toco -- 
 --input_file=/external_home/data/model/deeplabv3_mnv2_pascal_train_aug/frozen_inference_graph.pb 
 --output_file=/external_home/data/model/deeplabv3_mnv2_pascal_train_aug/kanul.tflite 
 --inference_type=QUANTIZED_UINT8 
 --input_shape=1,513,513,3 
 --input_array=sub_7 
 --output_array=logits/semantic/BiasAdd

Unimplemented: this graph contains an operator of type SpaceToBatchND for which the quantized form is not yet implemented. Sorry, and patches welcome (that's a relatively fun patch to write, mostly providing the actual quantized arithmetic code for this op).

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Exact command to reproduce

**Comment by kanul** [comment]
@tensorflowbutler @skye

"Exact command to reproduce" was updated. Could you check out that?

**Comment by kanul** [comment]
@suharshs

"Exact command to reproduce" was updated. Could you check out that?

**Comment by suharshs** [comment]
Hi @kanul ,

It seems that the error is that TOCO doesn't yet support quantized SpaceToBatchND. When i look at the kernel implementations, it seems that the kernel does support quantization but the transformation missing that. I will work on a fix soon.

Thanks!

**Comment by kanul** [comment]
@suharshs

And, can you support not SpaceToBatchND but also ResizeBilinear?

**Comment by suharshs** [comment]
ResizeBilinear needs a quantized kernel implementation, I believe this is in the works and should be ready within a couple of weeks.

**Comment by PrinceP** [comment]
Hi @suharshs,
Can you also point out the kernel implementations you are looking for the SpaceToBatchND ?
Would be good understanding pointer. 
Thank you

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suharshs: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by kanul** [comment]
Hi @suharshs,

Can you share for me your status for implement of ResizeBilinear?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suharshs: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by WenguoLi** [comment]
Hi, @suharshs,
Are there any updates about this problem ?

**Comment by suharshs** [comment]
@PrinceP here is a pointer to the SpaceToBatchND kernel implementation https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/kernels/internal/reference/reference_ops.h#L3282 I made a recent change to add support for quantization for it.

I believe ResizeBilinear is supported now but we still need support for dilated depthwise conv. Thus i will mark this a duplicate of https://github.com/tensorflow/tensorflow/issues/20684.
